src/plugins/intel_npu/src/al/src/sync_infer_request.cpp

```diff
@@ -195,8 +200,12 @@ void SyncInferRequest::check_tensor(const ov::Output<const ov::Node>& port,
 void SyncInferRequest::check_tensors() const {
     const auto& inputs = _compiledModel->inputs();
     for (size_t i = 0; i < inputs.size(); i++) {
-        if (_userInputTensors.at(i)) {
-            check_tensor(inputs[i], _userInputTensors.at(i));
+        if (is_batched_input(i)) {
+            // in case of batched input, user tensor is not set
+            continue;
+        }
+        if (_userInputTensors.at(i).at(0)) {
```

Comment:
`_userInputTensors.at(i).at(0)` could be hard to read.

Maybe, we can make `_userInputTensors.at(i)` named? For example, as `batchedInput` or something like that.
For all cases, when we have `at(...).at(...)`

Reply:
Yup, changed that. there are still some calls. I'm looking at how to change them as well.

Reply:
changed all of them right now.